pyscript.core/esm/plugins.js

```diff
@@ -40,16 +43,68 @@ export const handlePlugin = (node) => {
                     config,
                 );
                 engine.then((runtime) => {
-                    const module = registry.get(type);
-                    onRuntimeReady(node, {
+                    const module = create(registry.get(type));
+                    const {
+                        onBeforeRun,
+                        onBeforeRunAsync,
+                        onAfterRun,
+                        onAfterRunAsync,
+                        onRuntimeReady,
+                        codeBeforeRunWorker,
+                        codeBeforeRunWorkerAsync,
+                        codeAfterRunWorker,
+                        codeAfterRunWorkerAsync,
+                    } = options;
+
+                    // patch sync
+                    for (const [name, [before, after]] of [
+                        ["run", [onBeforeRun, onAfterRun]],
+                    ]) {
+                        const method = module[name];
+                        module[name] = function (runtime, code) {
+                            if (before) before.call(this, resolved, node);
+                            const result = method.call(this, runtime, code);
+                            if (after) after.call(this, resolved, node);
+                            return result;
+                        };
+                    }
+
+                    // patch async
+                    for (const [name, [before, after]] of [
+                        ["runAsync", [onBeforeRunAsync, onAfterRunAsync]],
+                    ]) {
+                        const method = module[name];
+                        module[name] = async function (runtime, code) {
+                            if (before) await before.call(this, resolved, node);
+                            const result = await method.call(
+                                this,
+                                runtime,
+                                code,
```

Comment:
I think maybe if we remove the command it will put this in a single line? (just a small nit haha)

```suggestion
                                code
```

Reply:
haha fair enough 😄